src/api/app/jobs/update_released_binaries_job.rb

```diff
@@ -36,109 +36,107 @@ def perform(event_id)
   #   - sets BinaryRelease.obsolete_time
   #
   def update_binary_releases_for_repository(repository, new_binary_releases, time = Time.now)
-    # building a hash to avoid single SQL select calls slowing us down too much
+    # We record BinaryRelease attributes in this hash of hashes to be able to compare them later on.
+    # This is an optimization so we do not have to fetch BinaryRelease individually from the database.
+    # FIXME: This is exactly what ActiveRecord::Associations::CollectionProxy does no?
```

Comment:
no, activerecord has always full objects in memory

Reply:
You talk about putting the full record into the hash, I talk about not using the hash at all...